test/recipes/99-test_fuzz_asn1parse.t

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env perl
+# Copyright 2016-2020 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+use strict;
+use warnings;
+
+use OpenSSL::Test qw/:DEFAULT srctop_file/;
+use OpenSSL::Test::Utils;
+
+setup("test_fuzz_asn1parse");
+
+my @fuzzers = ('asn1parse');
+
+plan tests => scalar @fuzzers + 1; # one more due to below require_ok(...)
+
+require_ok(srctop_file('test','recipes','fuzz.pl'));
+
+&fuzz_tests(@fuzzers);
```

Comment:
Consider that there is only one fuzzer being tested, you could reduce everything from the `setup` call to:

``` perl
setup("test_fuzz_asn1parse");

plan tests => 2;

require_ok(srctop_file('test','recipes','fuzz.pl'));
fuzz_tests('asn1parse');
```

Furthermore, you could reduce the `fuzz_test` function to this:

```
sub fuzz_test {
    die "Too many arguments" if scalar @_ > 1;

    my $f = $_[0];
    my @dir = glob(srctop_dir('fuzz', 'corpora', $f));

    plan skip_all => "No directory fuzz/corpora/$f" unless @dir;
    plan tests => scalar @dir; # likely 1

    foreach (@dir) {
        ok(run(fuzz(["$f-test", $_])));
    }
}
```

and then rewrite the recipe like this:

``` perl
my $fuzzed = 'asn1parse';

setup("test_fuzz_$fuzzed");

plan tests => 2;

require_ok(srctop_file('test','recipes','fuzz.pl'));
subtest "Fuzzing $fuzzed" => \&fuzz_tests, $fuzzed;
```

That makes it visually more obvious how many and what tests are being run.

Reply:
Uhm, how does this version support the `asn1/0` syntax?

Reply:
Ah, hadn't noticed the rewrite.  How about this, then:

``` perl
sub fuzz_test {
    die "Too many arguments" if scalar @_ > 1;

    my $d = $_[0];
    my @tokens = split(/\//, $arg);
    my @dir = glob(srctop_dir('fuzz', 'corpora', @tokens));

    plan skip_all => "No directory fuzz/corpora/$d" unless @dir;
    plan tests => scalar @dir; # likely 1

    my $f = $tokens[0];
    foreach (@dir) {
        ok(run(fuzz(["$f-test", $_])));
    }
}
```

Reply:
`$arg` was supposed to be `$d` right?

I should have implemented this and thank to your idea there is no need to move files around in `fuzz/corpora/asn1/`.